src/content/1.7/development/components/grid/tutorials/work-with-grid-actions.md

```diff
@@ -3,4 +3,180 @@ title: How to work with Grid Actions?
+
+* Import;
+* Export;
+* Refresh;
+* Show SQL query;
+* Export to the SQL Manager;
+
+See code example from the Core below:
+
+```php
+<?php // src/Core/Grid/Definition/Factory/CustomerGridDefinitionFactory.php
+
+final class CustomerGridDefinitionFactory extends AbstractGridDefinitionFactory
+{
+    // ...
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getGridActions()
+    {
+        return (new GridActionCollection())
+            ->add(
+                (new LinkGridAction('import'))
+                ->setName($this->trans('Import', [], 'Admin.Actions'))
+                ->setIcon('cloud_upload')
+                ->setOptions([
+                    'route' => 'admin_import',
+                    'route_params' => [
+                        'import_type' => 'customers',
+                    ],
+                ])
+            )
+            ->add(
+                (new LinkGridAction('export'))
+                ->setName($this->trans('Export', [], 'Admin.Actions'))
+                ->setIcon('cloud_download')
+                ->setOptions([
+                    'route' => 'admin_customers_export',
+                ])
+            )
+            ->add(
+                (new SimpleGridAction('common_refresh_list'))
+                ->setName($this->trans('Refresh list', [], 'Admin.Advparameters.Feature'))
+                ->setIcon('refresh')
+            )
+            ->add(
+                (new SimpleGridAction('common_show_query'))
+                ->setName($this->trans('Show SQL query', [], 'Admin.Actions'))
+                ->setIcon('code')
+            )
+            ->add(
+                (new SimpleGridAction('common_export_sql_manager'))
+                ->setName($this->trans('Export to SQL Manager', [], 'Admin.Actions'))
+                ->setIcon('storage')
+            );
+    }
+}
+```
+
+As you see, Grid stores Gird actions using a `GridActionCollection` object. This means that every action needed for Grid must be added to `GridActionCollection` using `add()` method.
+You'll find the list of existing actions in the documentation.
+
+## How to add Bulk actions to a Grid using a module?
+
+You can add additional actions to any Grid by hooking to the Grid workflow. You have to follow these steps:
+
+1. Register hook `action{gridId}GridDefinitionModifier` where `{gridId}` is Grid id (e.g. `customer` for Customers grid, this means that hook name would be `actionCustomerGridDefinitionModifier`).
+2. Implement hook method in your module:
+
+```php
+<?php
+// modules/my-module/my-module.php
+
+class Mymodule extends Module
+{
+    // ...
+
+    /**
+     * Use hook to add Bulk action for subscribing multiple customers to newsletter
+     */
+    public function hookActionCustomerGridDefinitionModifier(array $params)
+    {
+        // $params['definition'] is instance of \PrestaShop\PrestaShop\Core\Grid\Definition\GridDefinition
+        $params['definition']->getActions()->add(
```

Comment:
more important: it's ``getGridActions()``

Reply:
People will copy & paste

Reply:
it's even more important then